### PR TITLE
Robustness plugin

### PR DESCRIPTION
**Pull request by iamksuresh**
## Description

1. main.py as the main entry point
2. moved test assests to stock-plugins root (stock-plugins/user_defined_files)
3. refactored the folder and files structure.
4. Users can now execute the plugin as a module
5. Plugin will now accept comma separated assets paths (data,model,ground_truth), model classifications, ground truth value,plugin args

## Type of Change

refactor: Code refactoring without changes in functionality 

 perf: Performance improvements

## How to Test

Execute the below commands to run unit tests -
```
cd aiverify/stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/
pytest .

```

image

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x]  I have tested the changes locally and verified that they work as expected.
- [x]  I have added or updated the necessary documentation (README, API docs, etc.).
- [x]  I have added appropriate unit tests or functional tests for the changes made.
- [x]  I have followed the project's coding conventions and style guidelines.
- [x]  I have rebased my branch onto the latest commit of the main branch.
- [x]  I have squashed or reorganized my commits into logical units.
- [x]  I have added any necessary dependencies or packages to the project's build configuration.
- [x]  I have performed a self-review of my own code.
- [x]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>

### CODE REVIEW

File: stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/aiverify_robustness_toolbox/__main__.py
Comment (timlrx) [inline stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/aiverify_robustness_toolbox/__main__.py:1]: Running a test creates an output folder with `widgets/images/adv_samples` and `widgets/images/org_samples`. Both folders are empty. Can you check if that is correct? If so, do we actually need the folders?

File: stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/tests/e2e/test_e2e.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    ),
+    "ground_truth_path": str(
+        "../../../user_defined_files/data/sample_reg_donation_data.sav"
+    ),
+    "run_pipeline": False,
+    "model_type": ModelType.REGRESSION,
+    "ground_truth": "donation",
+}
+
+image_pipeline = {
```
Comment (timlrx) [inline stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/tests/e2e/test_e2e.py:33]: This is the result of the image pipeline e2e test:

```
{
    "results": {
        "num_of_perturbed_samples": 1,
        "org_performance": 1.0,
        "perturbed_performance": 1.0,
        "num_of_failed_perturbed_samples": 1,
        "perturbed_samples": [],
        "org_samples": [],
        "feature_names": [],
        "perturbed_pred": [
            9
        ],
        "original_pred": [
            [
                9
            ]
        ]
    }
}
```

It does not look very meaningful to me. Can you check if that's expected? The results for the tabular case seem fine.

File: stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/README.md [hunk 2]
```diff
@@ -1,10 +1,54 @@
...
+root_path="<PATH_TO_FOLDER>/aiverify/stock-plugins/user_defined_files"
+python -m aiverify_robustness_toolbox \
+  --data_path $root_path/data/sample_bc_pipeline_credit_data.sav \
+  --model_path $root_path/pipeline/bc_tabular_credit \
+  --ground_truth_path $root_path/data/sample_bc_pipeline_credit_ytest_data.sav \
+  --ground_truth default \
+  --model_type CLASSIFICATION \
+  --run_pipeline \
+  --annotated_ground_truth_path $root_path/data/pickle_pandas_fashion_mnist_annotated_labels_10.sav \
+  --file_name_label file_name
```
Comment (timlrx) [inline stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/README.md:40]: These two arguments don't have any use in the tabular example and we can exclude them

File: stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/aiverify_robustness_toolbox/__main__.py
Comment (iamksuresh) [reply stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/aiverify_robustness_toolbox/__main__.py:1]: @imda-peckyoke : do you know why we have the widget folders created ?

File: stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/tests/e2e/test_e2e.py (hunk 1 above)
Comment (iamksuresh) [reply stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/tests/e2e/test_e2e.py:33]: as discussed, based on output.schema.json only _["num_of_perturbed_samples", "org_performance", "perturbed_performance", "num_of_failed_perturbed_samples"]_ are the required fields.

**Review by imda-peckyoke — APPROVED** [review]
approve

File: stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/aiverify_robustness_toolbox/__main__.py
Comment (imda-peckyoke) [reply stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/aiverify_robustness_toolbox/__main__.py:1]: No idea. but if it doesn't impact anything let's leave it as it is first. Can resolve in later update.

Comment (iamksuresh) [reply stock-plugins/aiverify.stock.robustness-toolbox/algorithms/robustness_toolbox/aiverify_robustness_toolbox/__main__.py:1]: @imda-peckyoke : ok, you may then proceed to merge this PR.

**Pull request merged**